tests/foreman/maintain/test_upgrade.py

```diff
@@ -0,0 +1,163 @@
+"""Test class for satellite-maintain upgrade functionality
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ForemanMaintain
+
+:Assignee: gtalreja
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+
+from robottelo.logging import logger
+
+pytestmark = pytest.mark.destructive
+
+
+@pytest.mark.include_capsule
+def test_positive_satellite_maintain_upgrade_list(sat_maintain):
+    """List versions this system is upgradable to
+
+    :id: 12efec41-4f09-4199-a20c-a4525e773b78
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain upgrade list-versions
+
+    :expectedresults: Versions system is upgradable to are listed.
+    """
+    xy_version = '.'.join(sat_maintain.version.split('.')[:2])
+    previous_xy_version = '.'.join(
+        [xy_version.split('.')[0], str(int(xy_version.split('.')[1]) - 1)]
```

Comment:
@vsedmik Yes, -1 logic won't work when we get 7.0, but that's not going to happen I believe 😄 
And I kept this logic because `version` property takes sat/cap versions from rpm itself, and for `os_version` we've x.y version and for sat/cap we get x.y.z version so not sure `Version` would help here.

Reply:
Seeing the discussion here, I immediately started thinking about Satellite Stream. If it becomes a thing, we should be careful about the `y` version.

Reply:
@ogajduse yes, these are upgrade tests for ystream/zstream, and we're not sure how upgrades will look like for satstream yet, we can visit this later if some change needed

Reply:
`Version` handles x.y.z too. But I agree we may not see 7.0 anytime soon and for the Stream... I have no idea how is the versioning planned there.

Reply:
@vsedmik Yes, right, no one knows about future versioning, so can we visit this later, and can resolve this thread?